services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/ProdDiagnosisKeyBundler.java

```diff
@@ -59,21 +63,52 @@ public ProdDiagnosisKeyBundler(DistributionServiceConfig distributionServiceConf
       KeySharingPoliciesChecker sharingPoliciesChecker) {
     super(distributionServiceConfig);
     this.sharingPoliciesChecker = sharingPoliciesChecker;
+    this.euPackageName = distributionServiceConfig.getEuPackageName();
+    this.applyPoliciesForAllCountries = distributionServiceConfig.getApplyPoliciesForAllCountries();
   }
 
   /**
-   * Initializes the internal {@code distributableDiagnosisKeys} map, grouping the diagnosis keys based on the
-   * country and by the date on which they may be distributed, while respecting the expiry and shifting policies.
+   * Initializes the internal {@code distributableDiagnosisKeys} map, grouping the diagnosis keys based on the country
+   * and by the date on which they may be distributed, while respecting the expiry and shifting policies.
    */
   @Override
   protected void createDiagnosisKeyDistributionMap(Collection<DiagnosisKey> diagnosisKeys) {
     this.distributableDiagnosisKeys.clear();
     Map<String, List<DiagnosisKey>> diagnosisKeysMapped = groupDiagnosisKeysByCountry(diagnosisKeys);
 
-    diagnosisKeysMapped.keySet().forEach(country -> populateDistributableDiagnosisKeys(diagnosisKeysMapped, country));
+    diagnosisKeysMapped.keySet().forEach(country -> {
+      if (!country.equals("DE") && !applyPoliciesForAllCountries) {
+        populateDistributableDiagnosisKeysWithoutPolicies(diagnosisKeysMapped, country);
+      } else {
+        populateDistributableDiagnosisKeysWithPolicies(diagnosisKeysMapped, country);
+      }
+    });
+    populateEuPackageWithDistributableDiagnosisKeys();
   }
 
-  private void populateDistributableDiagnosisKeys(Map<String, List<DiagnosisKey>> diagnosisKeysMapped, String country) {
+  private void populateEuPackageWithDistributableDiagnosisKeys() {
+    Map<LocalDateTime, Set<DiagnosisKey>> euPackage = new HashMap<>();
+
+    this.distributableDiagnosisKeys.forEach((country, diagnosisKeyMap) -> {
+      diagnosisKeyMap.forEach((distributionDateTime, diagnosisKeys) -> {
+        Set<DiagnosisKey> currentHourDiagnosisKeys = Optional
+            .ofNullable(euPackage.get(distributionDateTime))
+            .orElse(new HashSet<>());
+        currentHourDiagnosisKeys.addAll(diagnosisKeys);
+        euPackage.put(distributionDateTime, currentHourDiagnosisKeys);
+      });
+
+    });
+    Map<LocalDateTime, List<DiagnosisKey>> worldBucketList = new HashMap<>();
```

Comment:
Since we're trying to combine the keys of the different country packages into a single euPackage we have to avoid duplicated keys. Therefore, the euPackage is using a Set<DiagnosisKey>, but later on a List<DiagnosisKey> is expected again.